src/tls.c

```diff
@@ -4717,8 +4717,13 @@ int TLSX_UseSupportedCurve(TLSX** extensions, word16 name, void* heap)
     SupportedCurve* curve = NULL;
     int ret;
 
-    if (extensions == NULL)
+    if (extensions == NULL) {
         return BAD_FUNC_ARG;
+    }
+
+    if (! TLSX_KeyShare_IsSupported(name)) {
```

Comment:
Check the pre-def static at tls.c66. Currently only for TLS v1.3 and HAVE_SUPPORTED_CURVES. This should be expanded. You can reproduce with `./configure --disable-tls13`.

```
src/tls.c:4724:11: error: implicit declaration of function 'TLSX_KeyShare_IsSupported' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if (! TLSX_KeyShare_IsSupported(name)) {
          ^
```

Reply:
ah...Thats why jenkins is being so noisy! Thanks.